include/seqan3/range/views/pairwise_combine.hpp

```diff
@@ -639,7 +252,403 @@ class pairwise_combine_view : public std::ranges::view_interface<pairwise_combin
     underlying_range_type u_range{};
     //!\brief The cached iterator pointing to the last element of the underlying range.
     std::ranges::iterator_t<underlying_range_type> back_iterator{};
-};
+}; // class pairwise_combine_view
+
+/*!\brief The internal iterator type for pairwise_combine_view.
+ * \tparam range_type The type of the range this iterator is operating on.
+ *
+ * \details
+ *
+ * This iterator models the iterator category of the `underlying_iterator_type`. It maintains a pair of iterators on
+ * the underlying range that move over all pairwise combinations of elements. The end is reached, when the second
+ * iterator points to the end of the underlying range and the first iterator to the last element of the range.
+ * Also note that this iterator does not model [Cpp17Iterator](https://en.cppreference.com/w/cpp/named_req/Iterator),
+ * since it does not return a reference to the represented type but a prvalue.
+ * Thus this iterator might not be usable with some legacy algorithms of the STL. But it is guaranteed to work with
+ * the ranges algorithms.
+ */
+ template <std::ranges::view underlying_range_type>
+ //!\cond
+     requires std::ranges::forward_range<underlying_range_type> && std::ranges::common_range<underlying_range_type>
+ // !\endcond
+template <typename range_type>
+class pairwise_combine_view<underlying_range_type>::iterator_type :
+                                    public std::ranges::view_interface<pairwise_combine_view<underlying_range_type>>
```

Comment:
```suggestion
class pairwise_combine_view<underlying_range_type>::iterator_type
```
This fixes the ICE

the iterator should not inherit from the view interface!

Reply:
I didn't know this was a goal I wanted to achieve 😇